src/scep/Client/envelope.py

```diff
@@ -62,6 +62,22 @@ def encrypt(self, data, algorithm = None):
         else:
             raise ValueError('Unrecognised encryption algorithm ', algorithm)
 
+        if keyEncAlg == 'rsa':
+            self._KeyEncryptionAlgorithm = KeyEncryptionAlgorithm(
+                    {'algorithm': KeyEncryptionAlgorithmId(u'rsa')}
+                    ) 
+        if keyEncAlg == 'rsaes_oaep' :
```

Comment:
can we make it `elif`

Reply:
`algorithm` and `keyEncAlg` are not strictly related so two separate `if` are needed I think. Could you provide more details on the requested change?